dashboard/test/controllers/pd/workshop_material_orders_controller_test.rb

```diff
@@ -0,0 +1,107 @@
+require 'test_helper'
+class Pd::WorkshopMaterialOrdersControllerTest < ::ActionController::TestCase
+  self.use_transactional_test_case = true
+  setup_all do
+    @workshop = create :pd_ended_workshop, course: Pd::Workshop::COURSE_CSF
+    @teacher = create :teacher
+    @enrollment = create :pd_enrollment, :from_user, user: @teacher, workshop: @workshop
+  end
+
+  setup do
+    Pd::Enrollment.any_instance.stubs(completed_survey?: true)
+  end
+
+  test_redirect_to_sign_in_for :new, params: -> {{enrollment_code: @enrollment.code}}
+
+  test 'teacher without survey is redirected to survey' do
```

Comment:
What do you redirect to if the enrollment is for a course that has not sent the survey?